src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,189 @@
+            System.out.println("todo <your task here>");
+            printLine();
+            handleCommand();
+        } catch (InvalidCommandSeparatorException e) {
+            printLine();
+            System.out.println("OOPS! The deadline/event description must be separated from " +
+                    "the time using '/by' or '/at'. Please follow this format:");
+            System.out.println("deadline <your task here> /by <your deadline time>");
+            System.out.println("event <your task here> /at <your event time period>");
+            printLine();
+            handleCommand();
+        }
+    }
+
+    public void addDeadlineOrEventTask(String[] inputWords, String type)
+            throws EmptyCommandArgumentException, InvalidCommandSeparatorException {
+        // Throw exception where command argument is empty
+        if (inputWords.length < 2) {
+            throw new EmptyCommandArgumentException();
+        }
+
+        // Find separator index
+        int separatorIndex = -1;
+        for (int i = 1; i < inputWords.length; i++) {
+            if (inputWords[i].equals("/by") || inputWords[i].equals("/at")) {
+                separatorIndex = i;
+                break;
+            }
+        }
+
+        // Throw exception where separator is not found
+        if (separatorIndex == -1) {
+            throw new InvalidCommandSeparatorException();
+        }
+
+        // Set description
+        String description = inputWords[1];
+        for (int i = 2; i < separatorIndex; i++) {
+            description = description + " " + inputWords[i];
+        }
+
+        // Set deadline (by when) or event time (at what time)
+        String time = inputWords[separatorIndex + 1];
+        for (int i = separatorIndex + 2; i < inputWords.length; i++) {
+            time = time + " " + inputWords[i];
+        }
+
+        if (type.equals("deadline")) {
+            taskList[listSize] = new Deadline(description, time);
+        } else {
+            taskList[listSize] = new Event(description, time);
+        }
+
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskList[listSize]);
+        printLine();
+
+        listSize++;
+    }
+
+    public void addTodoTask(String[] inputWords) throws EmptyCommandArgumentException {
+        // Throw exception where command argument is empty
+        if (inputWords.length < 2) {
+            throw new EmptyCommandArgumentException();
+        }
+
+        String description = inputWords[1];
+        for (int i = 2; i < inputWords.length; i++) {
+            description = description + " " + inputWords[i];
+        }
+
+        taskList[listSize] = new Todo(description);
+
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskList[listSize]);
+        printLine();
+
+        listSize++;
```

Comment:
Consider creating classes to handle task manipulation and perhaps store your task list there